destination/writer/writer.go

```diff
@@ -61,22 +62,21 @@ func NewWriter(ctx context.Context, params Params) *Writer {
 func (w *Writer) InsertRecord(ctx context.Context, record sdk.Record) error {
 	action := record.Metadata[metadataAction]
 
-	// TODO: handle update and delete actions
-	switch action {
-	case actionInsert:
-		return w.insert(ctx, record)
-	default:
-		return w.insert(ctx, record)
+	if action == actionDelete {
+		return w.delete(ctx, record)
 	}
+
+	return w.upsert(ctx, record)
 }
 
 // Close closes the underlying db connection.
 func (w *Writer) Close(ctx context.Context) error {
 	return w.db.Close()
 }
 
-// insert is an append-only operation that doesn't care about keys.
-func (w *Writer) insert(ctx context.Context, record sdk.Record) error {
+// upsert inserts or updates a record. If the record.Key is not empty the method
+// will try to upsert the existing row, otherwise, it will plainly append a new row.
```

Comment:
Maybe you mean "upsert" => "update"

Reply:
I've updated the comment. Thank you for your sharp eye :)